GoogleSignIn/Tests/Unit/GIDSignInTest.m

```diff
@@ -674,10 +675,12 @@ - (void)testOAuthLogin_LoginHint {
 }
 
```

Comment:
I know that `GIDSignInTest` is cluttered and that `GIDSignIn` is a difficult class to test (we're working on this), but could you add a test that verifies the passed in `nonce` is matched in the `OIDAuthorizationResponse` we back in [GIDSignIn](https://github.com/google/GoogleSignIn-iOS/blob/b0bc583eb31df0354addb1e0df651fa6da1d386e/GoogleSignIn/Sources/GIDSignIn.m#L610)?

It'd be great to have a unit test that asserts: `XCTAssertEqualObjects(originalNonce, authorizationResponse.request.nonce)`.